app/Events/ChatMessageEvent.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Events;
+
+use App\Models\Chat\Message;
+use App\Transformers\Chat\MessageTransformer;
+use App\Transformers\UserCompactTransformer;
+use Illuminate\Broadcasting\Channel;
+use Illuminate\Contracts\Broadcasting\ShouldBroadcastNow;
+
+class ChatMessageEvent implements ShouldBroadcastNow
```

Comment:
at over half second for 10k users this may not quite be good to be run synchronously as part of the request...